packages/headless/src/controllers/core/search-box/headless-core-search-box.ts

```diff
@@ -0,0 +1,271 @@
+import {
+  defaultSearchBoxOptions,
+  SearchBoxOptions,
+  searchBoxOptionsSchema,
+} from './headless-core-search-box-options';
+import {
+  SuggestionHighlightingOptions,
+  Delimiters,
+  getHighlightedSuggestion,
+} from '../../../utils/highlight';
+import {
+  buildController,
+  Controller,
+} from '../../controller/headless-controller';
+import {CoreEngine} from '../../..';
+import {
+  ConfigurationSection,
+  QuerySection,
+  QuerySetSection,
+  QuerySuggestionSection,
+  SearchSection,
+} from '../../../state/state-sections';
+import {
+  configuration,
+  query,
+  querySet,
+  querySuggest,
+  search,
+} from '../../../app/reducers';
+import {loadReducerError} from '../../../utils/errors';
+import {randomID} from '../../../utils/utils';
+import {validateOptions} from '../../../utils/validate-payload';
+import {
+  registerQuerySetQuery,
+  updateQuerySetQuery,
+} from '../../../features/query-set/query-set-actions';
+import {
+  clearQuerySuggest,
+  FetchQuerySuggestionsActionCreatorPayload,
+  registerQuerySuggest,
+  selectQuerySuggestion,
+} from '../../../features/query-suggest/query-suggest-actions';
+import {SearchAction} from '../../../features/analytics/analytics-utils';
+import {prepareForSearchWithQuery} from '../../../features/search/search-actions';
+import {logQuerySuggestionClick} from '../../../features/query-suggest/query-suggest-analytics-actions';
+import {logSearchboxSubmit} from '../../../features/query/query-analytics-actions';
+import {QuerySuggestState} from '../../../features/query-suggest/query-suggest-state';
+import {AsyncThunkAction} from '@reduxjs/toolkit';
+
+export type {SearchBoxOptions, SuggestionHighlightingOptions, Delimiters};
+
+export interface CoreSearchBoxProps {
+  /**
+   * The `CoreSearchBox` controller options.
+   */
+  options?: SearchBoxOptions;
+
+  executeSearchActionCreator?: (
+    arg: SearchAction
+  ) => AsyncThunkAction<any, SearchAction, any>;
```

Comment:
Unfortunately `AnyAction` doesn't work because the `AsyncThunkAction<>` does not have a `type` attribute, which makes sense. Also, I could not find another type matching any async thunk action.